mlir/utils/jenkins/Jenkinsfile

```diff
@@ -256,47 +256,143 @@ pipeline {
+                                   -DCMAKE_INSTALL_PREFIX=${WORKSPACE}/MIOpen/build/MIOpenInstallDir
+                                   ''')
+                                   sh 'cd build; make -j $(nproc)'
+                                   sh 'cd build; make install'
+                               }
+                                stash name:"MIOpen-kernels",\
+                                   includes: "MIOpen/build/**"
+                           }
+                        } 
+                        stage("Build libMLIRMIOpen and MLIR") {
+                            steps {
+                                buildProject('libMLIRMIOpen', '-DBUILD_FAT_LIBMLIRMIOPEN=ON')
+                                cmake arguments: '--install . --component libMLIRMIOpen --prefix /opt/rocm',\
+                                installation: 'InSearchPath', workingDir: 'build'
+                            }
+                        }
+                        stage("Build MIOpen with libMLIRMIOpen") {
                             steps {
                                 dir('MIOpen') {
-                                    sh 'mkdir MIOpenInstallDir'
-                                    getAndBuildMIOpen('--prefix ./MIOpenDeps',
-                                                      '''-DMIOPEN_BACKEND=HIP
-                                    -DCMAKE_PREFIX_PATH="${WORKSPACE}/MIOpen/MIOpenDeps"
-                                    -DCMAKE_INSTALL_PREFIX=${WORKSPACE}/MIOpen/MIOpenInstallDir
-                                    ''')
-                                    sh 'cd build; make -j $(nproc)'
-                                    sh 'cd build; make install'
+                                getAndBuildMIOpen('--prefix ./MIOpenDeps',
+                                       '''-DMIOPEN_BACKEND=HIP -DMIOPEN_USE_MLIR=On 
+                                          -DCMAKE_PREFIX_PATH="${WORKSPACE}/MIOpen/MIOpenDeps"
+                                          -DMIOPEN_USER_DB_PATH=${WORKSPACE}/MIOpen/build/MIOpenUserDB
+                                       ''')
+                                //sh 'cd build; make -j $(nproc)'
+                                //sh 'cd build; make install'
                                 }
                             }
                         }
-                        stage("Build MLIR") {
+                        stage("Tune MLIR Kernels") {
                             steps {
-                                buildProject('mlir-miopen-driver mlir-rocm-runner ci-performance-scripts', '')
+                                buildProject('ci-performance-scripts', '')
+                                dir('build') {
+                                    sh 'python3 ./bin/MIOpenDriver.py -tuning'
+                                }
+                                stash name:"MIOpen-MLIR-kernels",\
+                                    includes: "MIOpen/build/**"
```

Comment:
~~Don't you need to stash the perfdb, not the build?~~ Ok, looks like you put the perfdb together with the build result.

~~Also, it looks like we are doing tuning in two independent stages:~~
 - ~~Tune MLIR Kernels~~
 - ~~MIOpen resnet50 config test~~

On a second look this also seems fine: The tuning stage do not check correctness whereas the later one does.